ballerina/collection.bal

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2023 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+
+# Represents a MongoDB collection that can be used to perform operations on the collection.
+@display {
+    label: "MongoDB Collection"
+}
+public isolated client class Collection {
+
+    private final Database database;
+    private final string collectionName;
+
+    isolated function init(Database database, string collectionName) returns Error? {
+        self.database = database;
```

Comment:
Are we using the `database` parameter any other place rather than the init phase ? If not we do not need to keep this right ?